providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java

```diff
@@ -133,22 +146,12 @@ private <T> ProviderEvaluation<T> resolve(Class<T> type, String key, T defaultVa
 
         // missing flag
         if (flag == null) {
-            return ProviderEvaluation.<T>builder()
-                    .value(defaultValue)
-                    .reason(Reason.ERROR.toString())
-                    .errorCode(ErrorCode.FLAG_NOT_FOUND)
-                    .errorMessage(String.format("requested flag could not be found: %s", key))
-                    .build();
+            throw new FlagNotFoundError("flag: " + key + " not found");
```

Comment:
This IMO is one of the hidden features of the SDK. While throwing exceptions is not my first choice, using standard exceptions from SDK is acceptable.

Reply:
Ya there's a bit of performance cost, but I think that these are truly "Exceptional" situations: ie: not parts of "normal" control flow. If you are evaluating a missing flag or a flag of the wrong type, it's probably an error you should fix.

Reply:
There's one aspect we should remember, SDK log the error [^1]. If a flag configuration is incorrect (invalid targeting rule), then exception usage can result in multiple log entries 🤔 

[^1]: https://github.com/open-feature/java-sdk/blob/main/src/main/java/dev/openfeature/sdk/OpenFeatureClient.java#L140

Reply:
That's a good point.

Reply:
we could decide on the strategy we choose here - throw exception and let sdk handle it (convenient, less control) OR generate the response by ourselves (manual handling, more control) :)

Reply:
I think the error messages are clear enough without logging in the provider. We can just use `log.error` for stuff that only the provider understands (ie: grpc connection lost).

Reply:
Sure, let's do that. Will resolve the conversation with this agreemnet